### ISSUE TITLE
TypeError: 'NoneType' object is not subscriptable for older legacy release identifiers

### ISSUE THREAD

**Issue by fridex** [thoth-station/adviser#1664]
**Describe the bug**

See error log from prod deployment as reported by Landon:

```
2021-02-10 15:04:55,153 thoth.adviser.sieves.solved WARNING: Removing package ('pandas', '0.1', 'https://pypi.org/simple') due to build time error in the software environment - see https://thoth-station.ninja/j/buildtime_error
2021-02-10 15:04:55,153 thoth.adviser.run           ERROR: The resolution failed as an error was encountered: 'NoneType' object is not subscriptable
Traceback (most recent call last):
  File "/opt/app-root/src/thoth/adviser/run.py", line 75, in subprocess_run
    report: Union[DependencyMonkeyReport, Report] = resolver.resolve(
  File "/opt/app-root/src/thoth/adviser/resolver.py", line 1292, in resolve
    for state in self._do_resolve_states(with_devel=with_devel, user_stack_scoring=user_stack_scoring):
  File "/opt/app-root/src/thoth/adviser/resolver.py", line 1222, in _do_resolve_states
    for final_state in self._do_resolve_states_raw(
  File "/opt/app-root/src/thoth/adviser/resolver.py", line 1128, in _do_resolve_states_raw
    state = self._prepare_initial_state(with_devel=with_devel)
  File "/opt/app-root/src/thoth/adviser/resolver.py", line 719, in _prepare_initial_state
    package_versions = list(self._run_sieves(package_versions))
  File "/opt/app-root/src/thoth/adviser/resolver.py", line 303, in _run_sieves
    yield from result
  File "/opt/app-root/src/thoth/adviser/sieves/solved.py", line 72, in run
    for package_version in package_versions:
  File "/opt/app-root/src/thoth/adviser/sieves/index_enabled.py", line 56, in run
    for package_version in package_versions:
  File "/opt/app-root/src/thoth/adviser/sieves/locked.py", line 62, in run
    for package_version in package_versions:
  File "/opt/app-root/src/thoth/adviser/sieves/prereleases.py", line 58, in run
    for package_version in package_versions:
  File "/opt/app-root/src/thoth/adviser/sieves/pandas/py36_drop.py", line 78, in run
    if package_version.semantic_version.release[:2] < (1, 2):
TypeError: 'NoneType' object is not subscriptable
2021-02-10 15:04:55,198 thoth.adviser.run           ERROR: Child exited with exit code 512
```

Packages in Pipfile:

```
[packages]
boto3 = "*"
kafka-python = "*"
matplotlib = "*"
numpy = "*"
pandas = "*"
scipy = "*"
tensorflow = "*"
```

**To Reproduce**
1. Go to prod deployment
2. Ask for an advise for the given stack
3. See the error

**Expected behavior**

Adviser should not fail and an advise should be propagated back to the user.

**Comment by fridex** [comment]
None is not reported for any release (releases obtained from PyPI), might be we have some specific release in the database (?):

```python
>>> from packaging.version import parse as parse_version
>>> versions
['0.1', '0.10.0', '0.10.1', '0.11.0', '0.12.0', '0.13.0', '0.13.1', '0.14.0', '0.14.1', '0.15.0', '0.15.1', '0.15.2', '0.16.0', '0.16.1', '0.16.2', '0.17.0', '0.17.1', '0.18.0', '0.18.1', '0.19.0', '0.19.0rc1', '0.19.1', '0.19.2', '0.2', '0.20.0', '0.20.0rc1', '0.20.1', '0.20.2', '0.20.3', '0.21.0', '0.21.0rc1', '0.21.1', '0.22.0', '0.23.0', '0.23.0rc2', '0.23.1', '0.23.2', '0.23.3', '0.23.4', '0.24.0', '0.24.0rc1', '0.24.1', '0.24.2', '0.25.0', '0.25.0rc0', '0.25.1', '0.25.2', '0.25.3', '0.2b1', '0.2beta', '0.3.0', '0.3.0.beta', '0.3.0.beta2', '0.4.0', '0.4.1', '0.4.2', '0.4.3', '0.5.0', '0.6.0', '0.6.1', '0.7.0', '0.7.0rc1', '0.7.1', '0.7.2', '0.7.3', '0.8.0', '0.8.0rc1', '0.8.0rc2', '0.8.1', '0.9.0', '0.9.1', '1.0.0', '1.0.0rc0', '1.0.1', '1.0.2', '1.0.3', '1.0.4', '1.0.5', '1.1.0', '1.1.0rc0', '1.1.1', '1.1.2', '1.1.3', '1.1.4', '1.1.5', '1.2.0', '1.2.0rc0', '1.2.1', '1.2.2']
>>> pprint([parse_version(v).release for v in versions])
[(0, 1),
 (0, 10, 0),
 (0, 10, 1),
 (0, 11, 0),
 (0, 12, 0),
 (0, 13, 0),
 (0, 13, 1),
 (0, 14, 0),
 (0, 14, 1),
 (0, 15, 0),
 (0, 15, 1),
 (0, 15, 2),
 (0, 16, 0),
 (0, 16, 1),
 (0, 16, 2),
 (0, 17, 0),
 (0, 17, 1),
 (0, 18, 0),
 (0, 18, 1),
 (0, 19, 0),
 (0, 19, 0),
 (0, 19, 1),
 (0, 19, 2),
 (0, 2),
 (0, 20, 0),
 (0, 20, 0),
 (0, 20, 1),
 (0, 20, 2),
 (0, 20, 3),
 (0, 21, 0),
 (0, 21, 0),
 (0, 21, 1),
 (0, 22, 0),
 (0, 23, 0),
 (0, 23, 0),
 (0, 23, 1),
 (0, 23, 2),
 (0, 23, 3),
 (0, 23, 4),
 (0, 24, 0),
 (0, 24, 0),
 (0, 24, 1),
 (0, 24, 2),
 (0, 25, 0),
 (0, 25, 0),
 (0, 25, 1),
 (0, 25, 2),
 (0, 25, 3),
 (0, 2),
 (0, 2),
 (0, 3, 0),
 (0, 3, 0),
 (0, 3, 0),
 (0, 4, 0),
 (0, 4, 1),
 (0, 4, 2),
 (0, 4, 3),
 (0, 5, 0),
 (0, 6, 0),
 (0, 6, 1),
 (0, 7, 0),
 (0, 7, 0),
 (0, 7, 1),
 (0, 7, 2),
 (0, 7, 3),
 (0, 8, 0),
 (0, 8, 0),
 (0, 8, 0),
 (0, 8, 1),
 (0, 9, 0),
 (0, 9, 1),
 (1, 0, 0),
 (1, 0, 0),
 (1, 0, 1),
 (1, 0, 2),
 (1, 0, 3),
 (1, 0, 4),
 (1, 0, 5),
 (1, 1, 0),
 (1, 1, 0),
 (1, 1, 1),
 (1, 1, 2),
 (1, 1, 3),
 (1, 1, 4),
 (1, 1, 5),
 (1, 2, 0),
 (1, 2, 0),
 (1, 2, 1),
 (1, 2, 2)]
```

The legacy version issue is relevant though.

**Comment by fridex** [comment]
Reopening so we have this tracked until we have a new release of thoth-adviser for this.

**Comment by fridex** [comment]
The fix is present in the recent v0.25.0 release. It will be delivered in the next release cycle - https://github.com/thoth-station/thoth-application/pull/932
